src/AutoRest.CSharp/Common/Output/Expressions/Azure/AzureExtensibleSnippets.AzureRestOperationsSnippets.cs

```diff
@@ -49,6 +50,8 @@ public override TypedValueExpression GetTypedResponseFromBinaryData(Type respons
                 }
                 return responseType == typeof(BinaryData)
                     ? ResponseExpression.FromValue(rawResponse.Content, rawResponse)
+                    : responseType == typeof(AzureLocation) ?
+                    ResponseExpression.FromValue(New.Instance(typeof(AzureLocation), new[] { rawResponse.Content.ToObjectFromJson(typeof(string)) }), rawResponse)
```

Comment:
So this is for return type "AzureLocation"? Do you have a test?

Reply:
yes. I have write the code and test it.